woocommerce/woocommerce
# Filter is_purchasable when getting cart item from session

### All Submissions:






### Changes proposed in this Pull Request:

Introduce `woocommerce_cart_item_is_purchasable ` filter and provide access to the cart item data array. 

### How to test the changes in this Pull Request:

1. add this snippet to `functions.php`
```
function kia_filter_product_in_cart( $is_purchasable, $cart_key, $cart_item_data ) {
    if( $is_purchasable && isset( $cart_item_data['testing'] ) ) {
    	$is_purchasable = false;
    }
    return $is_purchasable;
}
add_filter( 'woocommerce_cart_item_is_purchasable', 'kia_filter_product_in_cart', 10, 3 );


function kia_add_test_item_data( $cart_item_data ) {
    $cart_item_data['testing'] = 1;
    return $cart_item_data;
}
add_filter( 'woocommerce_add_cart_item', 'kia_add_test_item_data' );
```
2. Add a simple product to the cart
3. Go to cart, the product should have been removed. 

This isn't a very practical test, but it shows it works. The use-case I'm envisioning is that Mix and Match and Bundles sometimes need to sell products _only_ as part of the bundle... ie: not sold separately. It's already possible to manipulate the purchasable status inside the bundle and both Mix and Match and Bundles add the bundled products to the cart, bypassing the default Woo process. However, when the cart is loaded from the session we hit this roadblock. Without access to the cart item data, there's nothing to use to filter `is_purchasable()` and report that this product is actually purchasable when being ordered as part of a bundle.


### Other information:




### Changelog entry

> New: `woocommerce_cart_item_is_purchasable ` filter

cc: @franticpsyx
